aws-cli/aws-cli.yaml

```diff
@@ -0,0 +1,30 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: aws-cli
+spec:
+  workspaces:
+    - name: source
+  params:
+    - name: SCRIPT
+      description: The AWS script to run
+      type: string
+      default: "aws $@"
+    - name: ARGS
+      description: AWS cli arguments to be passed
+      type: array
+      default: ["help"]
+  steps:
+    - name: awscli
+      image: amazon/aws-cli:latest
+      workingDir: "/tekton/home/.aws"
+      script: "$(params.SCRIPT)"
```

Comment:
can you explain more about why someone would want to override SCRIPT? with the user being able to override both SCRIPT and ARGS it seems like they could use this task to generically execute _anything_ inside the aws-cli image - maybe that's okay, but its a different choice than we've made with other Tasks where we try to make the scope more constrained

my recommendation here would be to at least always be executing the `aws` command - but if you feel strongly we can leave it like this!

Reply:
Yes, I tried to generalize the task here, so that we can run any `aws` commands.
Also, I took reference from existing `openshift-client` task. https://github.com/tektoncd/catalog/blob/v1beta1/openshift-client/openshift-client-task.yaml

cc @nikhil-thomas

Reply:
Right, it's similar to `openshift-client`, a very "generic" Task. Allowing writing their own scripts make the task more powerful.

Default it will do `aws help`, if you give args ("foo bar") it will do `aws foo bar`. But if you need something more complex you could do

```yaml
params:
- name: SCRIPT
  value: |
#!/usr/bin/env bash
while getopts u:d:p:f: option
do
case "${option}"
in
p) PREINIT=${OPTARG};;
n) NEXT=${OPTARG};;
esac
done
if [[ -z "$PREINIT" ]]; do 
    # run preinit
    $PRINIT
fi
echo "Do something"
aws subcommand $NEXT
echo "Do the rest"
aws anothersubcommand "$@"
- name: ARGS
  value: ["-p", "$(params.preinit)", "s3", "upload"]
```

Reply:
Here, I have added `SCRIPT` param to support multiple aws commands (as, in ARGS we can only pass single command). It will help us to avoid creating pipeline for running multiple commands. We can run all commands having only single taskrun.

Will add more explanation for this in README.